code/game/dna/dna_modifier.dm

```diff
@@ -533,7 +533,7 @@
 		occupantData["name"] = connected.occupant.dna.real_name
 		occupantData["stat"] = connected.occupant.stat
 		occupantData["isViableSubject"] = 1
-		if((NOCLONE in connected.occupant.mutations && connected.scan_level < 3) || !src.connected.occupant.dna || connected.occupant.dna.species.bodyflags & NO_DNA)
+		if((NOCLONE in connected.occupant.mutations && connected.scan_level < 3) || !src.connected.occupant.dna || NO_DNA in connected.occupant.dna.species.species_traits)
```

Comment:
Careful: `NO_DNA in ...` should be in parentheses, otherwise order of operations is going to trample all over this statement and make you have a bad day.